### ISSUE TITLE
TraceEvent Stack Traces not working for x64 Processes

### ISSUE THREAD

**Issue by Alois-xx** [Microsoft/perfview#332]
I have tried the TraceLogMonitor sample which is a great tool for real time exception tracing. But unfortunately it looks like TraceEvent is only capable to decode stack traces from x86 processes. Stack traces from x64 stacks show up as empty. 

Am I missing something obvious? 
The stacks for x64 look like 
`System.IO.FileNotFoundException: Could not load file or assembly 'MemoScope.XmlSerializers' or one of its dependencies. The system cannot find the file specified.

`
Whereas for x86
```
System.IO.FileNotFoundException: Could not load file or assembly 'MemoScope.XmlSerializers' or one of its dependencies. The system cannot find the file specified.
        EtwCallout
        CoTemplate_zzpqhh
        ETW::ExceptionLog::ExceptionThrown
        COMPlusThrowCallback
        Thread::StackWalkFramesEx
        Thread::StackWalkFrames
        CPFH_RealFirstPassHandler
        CPFH_FirstPassHandler
        _COMPlusFrameHandler
        _COMPlusNestedExceptionHandler
        ExecuteHandler2
        ExecuteHandler
        KiUserExceptionDispatcher
        RaiseTheExceptionInternalOnly
        UnwindAndContinueRethrowHelperAfterCatch
        AssemblyNative::Load
        System.Reflection.RuntimeAssembly.nLoad(System.Reflection.AssemblyName, System.String, System.Security.Policy.Evidence, System.Reflection.RuntimeAssembly, System.Threading.Sta
        System.Reflection.RuntimeAssembly.LoadWithPartialNameInternal(System.Reflection.AssemblyName, System.Security.Policy.Evidence, System.Threading.StackCrawlMark ByRef)
        System.Reflection.Assembly.LoadWithPartialName(System.String, System.Security.Policy.Evidence)
        System.Xml.Serialization.TempAssembly.LoadGeneratedAssembly(System.Type, System.String, System.Xml.Serialization.XmlSerializerImplementation ByRef)
        System.Xml.Serialization.XmlSerializer..ctor(System.Type, System.String)
        System.Xml.Serialization.XmlSerializer..ctor(System.Type)
        CallDescrWorkerInternal
        CallDescrWorkerWithHandler
        DispatchCallDebuggerWrapper
        DispatchCallSimple
        MethodTable::RunClassInitEx
        MethodTable::DoRunClassInitThrowing
        CEEInfo::initClass
        Compiler::impInitClass
        Compiler::impImportBlockCode
        Compiler::impImportBlock
        Compiler::impImport
        Compiler::compCompile
        Compiler::compCompileHelper
        Compiler::compCompile
        jitNativeCode
        CILJit::compileMethod
        invokeCompileMethodHelper
        invokeCompileMethod
        CallCompileMethodWithSEHWrapper
        UnsafeJitFunction
        MethodDesc::MakeJitWorker
        MethodDesc::DoPrestub
        PreStubWorker
        ThePreStub
        MemoScope.Modules.Explorer.FileData..ctor(class System.IO.FileInfo)
        MemoScope.Modules.Explorer.AbstractDumpExplorerData.GetItems(class System.String)
        MemoScope.Modules.Explorer.ExplorerModule.RefreshRootDir()
        MemoScope.Modules.Explorer.ExplorerModule.tbRootDir_TextChanged(class System.Object,class System.EventArgs)
        System.Windows.Forms.TextBoxBase.OnTextChanged(System.EventArgs)
        System.Windows.Forms.Control.set_Text(System.String)
        System.Windows.Forms.TextBoxBase.set_Text(System.String)
        System.Windows.Forms.TextBox.set_Text(System.String)
        [COLD] System.Windows.Forms.UserControl.OnLoad(System.EventArgs)
        System.Windows.Forms.UserControl.OnCreateControl()
        System.Windows.Forms.Control.CreateControl(Boolean)
        System.Windows.Forms.Control.CreateControl(Boolean)
        System.Windows.Forms.Control.CreateControl()
        System.Windows.Forms.Control.SetVisibleCore(Boolean)
        System.Windows.Forms.Form.SetVisibleCore(Boolean)
        System.Windows.Forms.Control.set_Visible(Boolean)
        WeifenLuo.WinFormsUI.Docking.DockContentHandler.SetVisible()
        WeifenLuo.WinFormsUI.Docking.DockPane.set_ActiveContent(class WeifenLuo.WinFormsUI.Docking.IDockContent)
        WeifenLuo.WinFormsUI.Docking.DockPane.ValidateActiveContent()
        WeifenLuo.WinFormsUI.Docking.DockContentHandler.SetDockState(bool,value class WeifenLuo.WinFormsUI.Docking.DockState,class WeifenLuo.WinFormsUI.Docking.DockPane)
        WeifenLuo.WinFormsUI.Docking.DockContentHandler.Activate()
        WeifenLuo.WinFormsUI.Docking.DockContentHandler.Show(class WeifenLuo.WinFormsUI.Docking.DockPanel)
        WeifenLuo.WinFormsUI.Docking.DockContentHandler.Show()
        WeifenLuo.WinFormsUI.Docking.DockContentHandler.Show(class WeifenLuo.WinFormsUI.Docking.DockPane,value class WeifenLuo.WinFormsUI.Docking.DockAlignment,float64)
        WeifenLuo.WinFormsUI.Docking.DockContent.Show(class WeifenLuo.WinFormsUI.Docking.DockPane,value class WeifenLuo.WinFormsUI.Docking.DockAlignment,float64)
        MemoScope.MemoScopeForm.MemoScope_Load(class System.Object,class System.EventArgs)

        [COLD] System.Windows.Forms.Form.OnLoad(System.EventArgs)
        System.Windows.Forms.Form.OnCreateControl()
        System.Windows.Forms.Control.CreateControl(Boolean)
        System.Windows.Forms.Control.CreateControl()
        System.Windows.Forms.Control.WmShowWindow(System.Windows.Forms.Message ByRef)
        System.Windows.Forms.Control.WndProc(System.Windows.Forms.Message ByRef)
        System.Windows.Forms.ScrollableControl.WndProc(System.Windows.Forms.Message ByRef)
        System.Windows.Forms.Form.WmShowWindow(System.Windows.Forms.Message ByRef)
        System.Windows.Forms.Form.WndProc(System.Windows.Forms.Message ByRef)
        System.Windows.Forms.Control+ControlNativeWindow.OnMessage(System.Windows.Forms.Message ByRef)
        System.Windows.Forms.Control+ControlNativeWindow.WndProc(System.Windows.Forms.Message ByRef)
        System.Windows.Forms.NativeWindow.Callback(IntPtr, Int32, IntPtr, IntPtr)

        _InternalCallWinProc
        UserCallWinProcCheckWow
        DispatchClientMessage
        __fnDWORD
        KiUserCallbackDispatcher
        System.Windows.Forms.Control.SetVisibleCore(Boolean)
        System.Windows.Forms.Form.SetVisibleCore(Boolean)
        System.Windows.Forms.Control.set_Visible(Boolean)
        System.Windows.Forms.Application+ThreadContext.RunMessageLoopInner(Int32, System.Windows.Forms.ApplicationContext)
        System.Windows.Forms.Application+ThreadContext.RunMessageLoop(Int32, System.Windows.Forms.ApplicationContext)
        System.Windows.Forms.Application.Run(System.Windows.Forms.Form)
        MemoScope.Program.Main(class System.String[])
        CallDescrWorkerInternal
```

I have used this code to print the stack traces
```
        static void Print(TraceEvent data, SymbolReader symbolReader)
        {
            // There are a lot of data collection start on entry that I don't want to see (but often they are quite handy
            if (data.Opcode == TraceEventOpcode.DataCollectionStart)
                return;
            // V3.5 runtimes don't log the stack and in fact don't event log the exception name (it shows up as an empty string)
            // Just ignore these as they are not that interesting. 
            if (data is ExceptionTraceData && ((ExceptionTraceData) data).ExceptionType.Length == 0)
                return;

            var ex = data as ExceptionTraceData;


            Out.WriteLine($"{ex.ExceptionType}: {ex.ExceptionMessage}");
            var callStack = data.CallStack();
            if (callStack != null)
            {
                // Because symbol lookup is complex, error prone, and expensive TraceLog requires you to be explicit.  
                // Here we look up names in this call stack using the symbol reader.  
                ResolveNativeCode(callStack, symbolReader);
                var current = callStack;
                while(current != null)
                {
                    Out.WriteLine($"\t{current.CodeAddress.FullMethodName}");
                    current = current.Caller;
                }
            }
        }


```

**Comment by Alois-xx** [comment]
I have found the issue. The pointersize of the trace session was wrong which did mess up the call stack table for x64 processes because the x64 addresses were treated as kernel addresses. I can create a PR if one cares.

**Comment by gleocadie** [comment]
I think it's a good idea to create a PR. Everyone will benefit from it ;)

**Comment by Alois-xx** [comment]
Fixed with PR #341
